python/python/bystro/supervised_ppca/tests/test_dropout_pt.py

```diff
@@ -0,0 +1,41 @@
+import numpy as np
+from sklearn.linear_model import LogisticRegression  # type: ignore
+from sklearn.metrics import roc_auc_score  # type: ignore
+from scipy.stats import ortho_group  # type: ignore
+from scipy.stats import binom  # type: ignore
+
+from bystro.supervised_ppca.gf_dropout_pt import PPCADropout
+
+
+def PPCA_generate_data(N=10000, L=5, p=30, phi=1.0, sigma=1.0):
+    rng = np.random.default_rng(2021)
+    S = rng.normal(size=(N, L))
+    W_o = ortho_group.rvs(p) * np.sqrt(p)
+    W = W_o[:L] * 3
+    W[0] = 0.3 * W[0]
+
+    X_hat = np.dot(S, W)
+    X_noise = sigma * rng.normal(size=(N, p))
+    X = X_hat + X_noise
+
+    logits = phi * S[:, 0] * 5
+    probs = np.exp(logits) / (1 + np.exp(logits))
+    y = binom.rvs(1, probs, random_state=42)
+    return X, y, X_hat, S, W, logits
+
+
+def test_ppca():
+    X, y, X_hat, S, W, logits = PPCA_generate_data(L=3, p=400)
+    training_options = {"n_iterations": 3000}
```

Comment:
(2)

Reply:
Sorry, my bad, on further reflection I don't think (2) is going to work, especially if we continue in this vein, because we're still going to have to run everything on CI.

Would you agree that it's a reasonable goal to have the test suite finish in a single-digit number of minutes?  Asking because we're almost hitting that ceiling now.

If we're going to do train for more than a single-digit number of iterations we should do so with a tiny model on a tiny dataset.  Would we lose substantive by reducing the number of latent dimensions to 1 or 2 here?

Reply:
The program part of unit tests takes 2.5 minutes, most of it I have no control over.